app/src/store/actions.ts

```diff
@@ -306,6 +306,10 @@ export async function searchBusiness ({ getters }, corpNum: string): Promise<Bus
   try {
     // first try to find business in Entities (Legal API)
     const data = await NamexServices.searchEntities(corpNum)
+    // Check if business is eligible for restoration NR by verifying restorationExpiryDate exists
+    if (getters.isRestoration && !data.restorationExpiryDate) {
```

Comment:
I think you should also check that the restoration expiry date isn't in the past.

I'm not sure whether the limited restoration expires on the date or the day before. Please look that up (maybe in Filings UI).

Reply:
Thanks for pointing this out. I've checked the restoration expiry date validation in [business-create-ui,](https://github.com/bcgov/business-create-ui/blob/main/src/mixins/amalgamation-mixin.ts#L254-L271).

I've updated.